### ISSUE TITLE
Enable Service Catalog standalone installation - phase 1

### ISSUE THREAD

**Issue by PK85** [kyma-project/kyma#2222]
**Description**

Make Service Catalog component e2e (backend, UI, installation process) modular, that means Service Catalog and all dependencies(are modular themselves as well) are installed only. That means to have a minimum setup in Kyma to be able to login and see only Service Catalog in UI and only custom resources from Service Catalog domain.

**Reasons**

1. Next phase of Kyma modularization.  It's the first step towards enablement only selected Kyma components in a standalone fully working solution.
2. It will enable to develop and run Kyma locally in a lightweight version and do not be disturbed by other components.
3. Will make contribution to Kyma easier (see point above). Will allow to install Service Catalog UI to any k8s cluster also with already installed Service Catalog.
4. Will make possible and easier to donate SC UI and SC addons to the kubernetes community in the future.

**Acceptance Criteria**
- Move Service Catalog Addons chart from core to root resources.
- List all components that are required to be installed to make Service Catalog and SC UI working (like dex, ui-api-layer), document that.
- Make instruction how to install Kyma without Service Catalog.
- document next step, to make our test to be modular as well (example: acceptance-test, kubeles test using SBU etc)

**Out of scope**
- UI will display all others component. But they will return 404. Next development phase will fix that.

**Attachments**

**Comment by adamwalach** [comment]
| Subtask | PRs |
| -------- | ---- |
| Move Service Catalog Addons chart from core to root resources. | https://github.com/kyma-project/kyma/pull/2515 |
List all components that are required to be installed to make Service Catalog and SC UI working (like dex, ui-api-layer), document that.| |
Make instruction how to install Kyma without Service Catalog.| |
document next step, to make our test to be modular as well (example: acceptance-test, kubeles test using SBU etc)| |

**Comment by adamwalach** [comment]
Additional acceptance criteria:
- move Helm Broker from core to root resources
- list dependencies to Service Catalog add-ons (we want to consolidate some of the modules)
- tests documentation - tests only per component, e2e should be triggered only during installation on cluster

**Comment by mszostok** [comment]
### General points for testing after modularization

TL;DR; 
- Component specific tests should be moved directly to the related component charts.
- Component tests should be more independent. Currently, e.g. Kubeless tests use the Helm Broker classes. Instead of that, they should register their own broker for that purpose.
- Component specific tests can required that their dependencies are installed. Cause there is no reason to e.g. test helm-broker without preinstalled service catalog.

- The integration tests which are testing multiple components should be placed in new chart called e.g. "acc-tests". 
```
.
├── CODEOWNERS
├── installation
└── resources
    ├── logging
    │   ├── Chart.yaml
    │   ├── README.md
    │   ├── charts
    │   ├── templates
    │   │   ├── _helpers.tpl
    │   │   ├── configmap.yaml
    │   │   ├── daemonset.yaml
    │   │   ├── service.yaml
    │   │   ├── statefulset.yaml
    │   │   └── tests
    │   │       └── logging.yaml              # Component test definition
    │   └── values.yaml
    ├── monitoring
    │   ├── Chart.yaml
    │   ├── charts
    │   ├── templates
    │   │   └── tests
    │   │       └── test-kube-prometheus.yaml # Component test definition
    │   └── values.yaml
    └── acc-tests                             # General E2E tests  
        ├── Chart.yaml
        ├── templates
        │   ├── _helpers.tpl
        │   └── tests
        │       ├── test-ui.yaml     
        │       └── test-app-binding.yaml
        └── values.yaml
```




### Investigation of current test from core
AI: 
- _TestApplicationMapping_EnsureBrokerAndClassesOnlyWhenMappingExist_ - move to **application-connector** chart.
- Dex: _TestSpec_ moved to **dex** chart.
- _TestServiceBindingUsagePrefixing_ - move to **service-catalog-addons** chart. Get rid of the ApplicationBroker domain here.
- _TestServiceCatalogContainsClusterServiceClasses_ - move to **helm-broker** chart.
- _TestServiceCatalogContainsABServiceClasses_ - move to **application-connector**  chart.
- [proposition] Test for ui-api-layer move to the the chart where ui-api will be placed. Change test runner and execute test only if the [BackendModule](https://github.com/kyma-project/kyma/blob/9b5583df4d151385e72b534ef91cddf0e3845d69/docs/console/docs/06-03-backendmodule.md) for tested domain is available.
- [proposition] Test for UI, move to the new chart called **acc-tests**.

---
**test-core-acceptance.yaml:**
- TestApplicationAPIAccess (domain: Istio, Service Catalog, Application Broker - currently disabled, see: https://github.com/kyma-project/kyma/issues/1453)
- TestApplicationMapping_EnsureBrokerAndClassesOnlyWhenMappingExist (domain: Service Catalog, ApplicationBroker)

- Dex: TestSpec (Should issue an ID token) runs only if `localEnv: true`

- TestServiceBindingUsagePrefixing (domain: ServiceCatalog, ApplicationBroker, BUC)
- TestServiceCatalogContainsClusterServiceClasses (domain: Service Catalog, Helm Broker)
- TestServiceCatalogContainsABServiceClasses (domain: Service Catalog, Application Broker)

**test-ui-api-acceptance.yaml**
- basically all test for ui-api-layer, but depends on various domains, e.g. Service Catalog: 
```
query ($namespace: String!) {
	serviceBrokers(namespace: $namespace) {
		%s
	}
}	
```

**test-ui-acceptance.yaml**
-  Test which covers the UI.
